pkg/internal/ebpf/common/kafka_detect_transform.go

```diff
@@ -182,31 +183,64 @@ func isValidClientID(buffer []byte, realClientIDSize int) bool {
 	return isValidKafkaString(buffer, len(buffer), realClientIDSize, true)
 }
 
-func getTopicName(pkt []byte, offset int) (string, error) {
+func getTopicName(pkt []byte, offset int, op Operation, apiVersion int16) (string, error) {
+	if apiVersion >= 13 { // topic name is only a UUID, no need to parse it
+		return "", nil
+	}
+
 	offset += 4
 	if offset > len(pkt) {
 		return "", errors.New("invalid buffer length")
 	}
-	topicNameSize := int16(binary.BigEndian.Uint16(pkt[offset:]))
-	if topicNameSize <= 0 || topicNameSize > 255 {
-		return "", errors.New("invalid topic name size")
+	topicNameSize, err := getTopicNameSize(pkt, offset, op, apiVersion)
+	if err != nil {
+		return "", err
 	}
 	offset += 2
 
 	if offset > len(pkt) {
 		return "", nil
 	}
-	maxLen := offset + int(topicNameSize)
+	maxLen := offset + topicNameSize
 	if len(pkt) < maxLen {
 		maxLen = len(pkt)
 	}
 	topicName := pkt[offset:maxLen]
+	if op == Fetch && apiVersion > 11 { // topic name has the following format: uuid\x00\x02\tTOPIC\x02\x00
+		topicNames := strings.Split(string(topicName), "\t")
+		if len(topicNames) >= 2 {
+			topicNames := strings.Split(string(topicNames[1]), "\x02\x00")
+			if len(topicNames) > 0 {
+				topicName = []byte(topicNames[0])
+			}
```

Comment:
Perhaps we can use trim of the \x02\x00 here, there's no else statement to handle the topic name.

Reply:
For some reason. `strings.TrimRight(string(topicNames[0]), "\\x02\x00")` is not doing anything, so that's why I had to do splits.

Reply:
Weird... I just did a simple test and worked: https://go.dev/play/p/LKz9ex2F5DH?v=

Don't forget to assign the result of `TrimRight` to a new variable.

Reply:
I ended using regexs. I think it's easier to understand and implementation is shorter